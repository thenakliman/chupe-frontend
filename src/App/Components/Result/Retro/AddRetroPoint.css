.add-retro-point-button {
    border: 1px solid #28a745;
    background-color: #28a745;
    padding: 0.8rem 1rem;
    color: white;
    background-image: linear-gradient(-180deg, #34d058 0%, #28a745 90%);
    cursor: pointer;
    margin: 2rem;
}

.add-retro-point-container {
    display: flex;
    flex-direction: column;
    background: #fff;
    padding: 4rem 3rem;
    border: 1px solid #ccc;
    margin: 7% 32%;
    position: fixed;
}

.add-retro-point-creation-input {
    padding: 0.8rem 0;
    width: 100%;
    font-size: 1.4rem;
    margin: 2rem 0;
}