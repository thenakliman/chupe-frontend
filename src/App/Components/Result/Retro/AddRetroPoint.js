import React from 'react';
import propTypes from 'prop-types';

require('./AddRetroPoint.css');

export class AddRetroPoint extends React.Component {
    constructor(props) {
      super(props);
      this.state = {description: ''};
      this.handleRetroDescription = this.handleRetroDescription.bind(this);
      this.createRetroPoint = this.createRetroPoint.bind(this);
    }

    handleRetroDescription(description) {
      this.setState({description: description});
    }

    createRetroPoint() {
      this.props.createRetroPoint(this.state.description);
      this.setState({description: ''});
    }

    render() {
      return (
      <div className='retro-pop-up-container'>
        <div id='add-retro-point-container-id'
             className='add-retro-point-container'>
                <input className={'add-retro-point-creation-input'}
                       id={'add-retro-point-description'}
                       onChange={(event) =>
                          this.handleRetroDescription(event.target.value)}
                       value={this.state.description}
                />
                <div className='add-retro-point-buttons-container'>
                    <button id='add-retro-point-button-id'
                      className='add-retro-point-button'
                      type='button'
                      onClick={this.createRetroPoint}
                    >
                      Add Retro Point
                    </button>
                    <button id='cancel-add-retro-point-button-id'
                      className='add-retro-point-button'
                      onClick={this.props.closeAddRetroPointPopUp}
                      type='button'
                    >
                      Cancel
                    </button>
                </div>
        </div>
      </div>);
    }
}

AddRetroPoint.propTypes = {
  closeAddRetroPointPopUp: propTypes.func.isRequired,
  createRetroPoint: propTypes.func.isRequired,
};
