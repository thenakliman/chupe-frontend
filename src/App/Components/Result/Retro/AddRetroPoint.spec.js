/* eslint-disable */
import React from 'react';
import {AddRetroPoint} from './AddRetroPoint';
/* eslint-enable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Show create retro pop up component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
        <AddRetroPoint
            createRetroPoint={jest.fn()}
            closeAddRetroPointPopUp={jest.fn()}
        />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have initial state for description and type', () => {
    const wrapper = shallow(
        <AddRetroPoint
            createRetroPoint={jest.fn()}
            closeAddRetroPointPopUp={jest.fn()}
        />);
    expect(wrapper.state()).toEqual({description: ''});
  });

  it('should call closeAddRetroPointPopUp on click of cancel button', () => {
    const wrapper = shallow(
        <AddRetroPoint
            createRetroPoint={jest.fn()}
            closeAddRetroPointPopUp={jest.fn()}
        />);

    wrapper.find('#cancel-add-retro-point-button-id').simulate('click');
  });

  it('should change state description state on edit', () => {
    const wrapper = shallow(
        <AddRetroPoint
            createRetroPoint={jest.fn()}
            closeAddRetroPointPopUp={jest.fn()}
        />);

    const description = 'description';
    wrapper.find('#add-retro-point-description').simulate('change',
        {target: {value: description}});

    expect(wrapper.state().description).toEqual(description);
  });

  it('should value of description on click of add retro point', () => {
    const wrapper = shallow(
        <AddRetroPoint
            createRetroPoint={jest.fn()}
            closeAddRetroPointPopUp={jest.fn()}
        />);

    wrapper.find('#add-retro-point-button-id').simulate('click');

    expect(wrapper.state().description).toEqual('');
  });

  it('should call retro point create on click of add retro point', () => {
    const createRetroPoint = jest.fn();
    const wrapper = shallow(
        <AddRetroPoint
            createRetroPoint={createRetroPoint}
            closeAddRetroPointPopUp={jest.fn()}
        />);

    wrapper.find('#add-retro-point-button-id').simulate('click');

    expect(createRetroPoint).toHaveBeenCalledWith('');
  });
});
