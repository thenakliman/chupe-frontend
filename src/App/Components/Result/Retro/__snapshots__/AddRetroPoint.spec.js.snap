// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show create retro pop up component should match snapshot 1`] = `
<div
  className="add-retro-point-container"
  id="add-retro-point-container-id"
>
  <input
    className="add-retro-point-creation-input"
    id="add-retro-point-description"
    onChange={[Function]}
    value=""
  />
  <div
    className="add-retro-point-buttons-container"
  >
    <button
      className="add-retro-point-button"
      id="add-retro-point-button-id"
      onClick={[Function]}
      type="button"
    >
      Add Retro Point
    </button>
    <button
      className="add-retro-point-button"
      id="cancel-add-retro-point-button-id"
      onClick={[MockFunction]}
      type="button"
    >
      Cancel
    </button>
  </div>
</div>
`;
